### PR TITLE
HMS-2861: Use RTK for state management

### PR DESCRIPTION
**Pull request by lucasgarfield**
*This is currently a WIP, there are a few bugs and some clean-up/refactoring is necessary*

So far we have considered (thanks to the work of @lavocatt) two possible options for managing the wizard state: storing state in the highest level component via useState() hooks, and storing state via a useContext() hook.

However, there is a 3rd option - we can use Redux (with RTK).

One big advantage is we can take advantage of the Redux dev tools and use them to inspect the state of the wizard during development:
image

### CODE REVIEW

**Comment by lavocatt** [comment]
Thanks for working on this other way of doing things. I'm still reviewing you PR and will hopefully soon be able to write you my thoughts about it.

**Comment by lavocatt** [comment]
I'm in the middle of the review now and I've noticed a bug. The wizard keeps it's state across multiple spawns. To reproduce the issue, open the wizard, fill up a couple of fields, click on cancel and open it up again. The fields that have been selected in the previous wizard session are still active. The expected behavior is that on every new click on `create image` the wizard starts with its fields set to the default values.

**Review by lavocatt — CHANGES_REQUESTED** [review]
Thanks @lucasgarfield this is very interesting approach and I'm glad you did the experimentation. The Redux toolkit is quite interesting, and the fact that we can explore the global state at any point in time adds value to this contribution.

In fairness to the other two approaches, the one with the `Context` and the one with the states lifted up to the wizard root, let's not forget that the React dev toolkit already allows us to explore the current wizard state:

For the `context`:

image

For the prop lifting:

image

One advantage as you mentioned on slack for the Redux toolkit is that it allows you to get back in time, one inconvenient is that you always have to scroll a lot to get to the latest version of the global state and it gets mixed up with the states of the RTK queries.

There still work for this PR to be a drop in replacement but you're getting there :rocket:

File: src/store/wizardSlice.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    architecture: X86_64,
+    targetEnvironment: [],
+  },
+};
+
+export const wizardSlice = createSlice({
+  name: 'wizard',
+  initialState,
+  reducers: {
+    changeRelease: (state, action) => {
```
[the remark this answers is not in the document]
Comment (lucasgarfield) [reply src/store/wizardSlice.ts:25]: They're reducers. They take a state and an action, and they return a state. In normal Redux state updates should be done immutably, but RTK uses the Immer library which allows us to write mutable state updates.

https://redux.js.org/tutorials/fundamentals/part-3-state-actions-reducers#writing-reducers

**Comment by lucasgarfield** [comment]
Closing this for a fresh start.

**Pull request closed without merging**